emmet-core/emmet/core/molecules/redox.py

```diff
@@ -0,0 +1,248 @@
+                # Start with highest lot; keep going down until you can make complete documents
+                for lot, group in sorted(lot_groups.items(), key=lambda x: evaluate_lot(x[0])):
+                    # Sorting important because we want to make docs only from lowest-energy instances
+                    ffopts = sorted(
+                        [f for f in group if f["task_type"] == TaskType.frequency_flattening_geometry_optimization],
+                        key=lambda x: x["output"]["final_energy"])
+
+                    charges = [f["charge"] for f in ffopts]
+                    if all([c in docs_by_charge for c in charges]):
+                        continue
+                    single_points = [s for s in group if s["task_type"] == TaskType.single_point]
+
+                    for ff in ffopts:
+                        d = {
+                            "electron_affinity": None,
+                            "ea_id": None,
+                            "ionization_energy": None,
+                            "ie_id": None,
+                            "reduction_free_energy": None,
+                            "red_id": None,
+                            "oxidation_free_energy": None,
+                            "ox_id": None,
+                            "reduction_potentials": None,
+                            "oxidations_potentials": None
+                        }
+
+                        charge = ff["charge"]
+
+                        # Doc already exists at a higher LOT; move on
+                        if charge in docs_by_charge:
+                            continue
+
+                        ff_mol = ff["output"]["optimized_molecule"]
+
+                        ff_g = get_free_energy(
+                            ff["output"]["final_energy"],
+                            ff["output"]["enthalpy"],
+                            ff["output"]["entropy"]
+                        )
+
+                        # Look for IE and EA SP
+                        for sp in single_points:
+                            sp_mol = sp["output"]["initial_molecule"]
+
+                            # EA
+                            if sp["charge"] == charge - 1 and mm.fit(ff_mol, sp_mol):
+                                d["electron_affinity"] = (sp["output"]["final_energy"] - ff["output"]["final_energy"]) * 27.2114
+                                d["ea_id"] = sp["task_id"]
+                            elif sp["charge"] == charge + 1 and mm.fit(ff_mol, sp_mol):
+                                d["ionization_energy"] = (sp["output"]["final_energy"] - ff["output"]["final_energy"]) * 27.2114
+                                d["ie_id"] = sp["task_id"]
+
+                            if d["ea_id"] is not None and d["ie_id"] is not None:
+                                break
+
+                        # If no vertical IE or EA, can't make complete doc; give up
+                        if d["ea_id"] is None or d["ie_id"] is None:
```

Comment:
I actually see it the opposite way. Because molecules might be unstable at different charges, or because the bonding might change, I expect there to be lots of gaps in the reduction and oxidation potentials. Even more generally, we just don't try all charges (a molecule at -1 in LIBE generally won't have a reduction potential because we don't calculate many molecules at -2).

On the other hand, vertical IE/EA calculations, in addition to being cheap, should pretty much always converge - they're single-points, after all. There might be some gaps, but yeah, pretty much everything in LIBE and MADEIRA has vertical IE and EA calculated.  If you don't have those, the most basic redox property that you could calculate, it seems not really worth it to even have a document. I guess there's an argument for making a document if you have any of these properties - IE or EA or reduction potential or oxidation potential. Would be willing to talk through that more.